injection/sharedmain/main.go

```diff
@@ -111,6 +113,38 @@ var (
 	WebhookMainWithConfig  = MainWithConfig
 )
 
+// MainNamed runs the generic main flow for controllers and webhooks.
+func MainNamed(ctx context.Context, component string, ctors ...injection.NamedControllerConstructor) {
+
+	disabledControllers := flag.String("disable-controllers", "", "Comma-separated list of disabled controllers.")
+
+	// HACK: This parses flags, so the above should be set once this runs.
+	cfg := injection.ParseAndGetRESTConfigOrDie()
+
+	enabledCtors := enabledControllers(strings.Split(*disabledControllers, ","), ctors)
+
+	MainWithConfig(ctx, component, cfg, enabledCtors...)
+}
+
+func enabledControllers(disabledControllers []string, ctors []injection.NamedControllerConstructor) []injection.ControllerConstructor {
+
+	disabledControllersSet := sets.NewString()
```

Comment:
seems like a shame not to not be able to use `sets.NewString(disabledControllers...)` here, do we definitely need to support spaces in the flag input? (looks like pflag.StringSlice doesn't strip spaces so it may make sense to stay consistent with that..)